stdlib/src/builtin/builtin_slice.mojo

```diff
@@ -16,27 +16,14 @@ These are Mojo built-ins, so you don't need to import them.
 """
 
 from sys.intrinsics import _mlirtype_is_eq
+from collections import OptionalReg
 
 
 @always_inline("nodebug")
-fn _int_max_value() -> Int:
-    # FIXME: The `slice` type should have the concept of `None` indices, but the
-    # effect of a `None` end index is the same as a very large end index.
-    return int(Int32.MAX)
-
-
-@always_inline("nodebug")
-fn _default_or[T: AnyRegType](value: T, default: Int) -> Int:
-    # TODO: Handle `__index__` for other types when we have traits!
-    @parameter
-    if _mlirtype_is_eq[T, Int]():
-        return __mlir_op.`kgen.rebind`[_type=Int](value)
-    else:
-        __mlir_op.`kgen.param.assert`[
-            cond = (_mlirtype_is_eq[T, NoneType]()).__mlir_i1__(),
-            message = "expected Int or NoneType".value,
-        ]()
-        return default
+fn _compare_optional(x: OptionalReg[Int], y: OptionalReg[Int]) -> Bool:
```

Comment:
Non-blocking: If you are interested, maybe we could add `__eq__` and `__ne__` to `OptionalReg` using the conditional conformance trick.